ejb/WFLY-13354-galleon-for-ejb.adoc

```diff
@@ -0,0 +1,115 @@
+This *could* depend on *ejb3* instead, but that means EJB is unnecessarily exposed over the HTTP interface and a lot of
+dependencies are brought in for messaging integration.
+
+All ejb layers should be preferring elytron as security provider, with an option to select legacy security.
+We should also make sure not to accidentally leave anything unsecured by default.
+
+Provisioning ejb3 + ejb3-iiop is similar to what's in standalone-full.xml, except that ejb3 subsystem in standard distribution
+uses legacy security domain.
+
+There'd be a couple ancillary layers as well:
+
+1. ejb-local-cache. This provides the infinispan subsystem resources related to local ejb caching
+(i.e. the ones in standalone.xml and standalone-full.xml.) The ejb3-lite layer *optionally* depends on this.
+It's an optional dependency because the user when provisioning can *exclude* this layer and instead provision...
+
+2. ejb-dist-cache. This provides the infinispan subsystem resources related to distributed ejb caching
+(i.e. the ones in standalone-ha.xml and standalone-full-ha.xml.)
+
+This pattern of "exclude an optional local caching layer and add a distributed variant" is how web session
+and jpa caching are handled in the layers for those features.
+
+TBH I don't see big FS footprint improvements coming out of these permutations. Big EJB dependencies include (in no particular order.)
+
+* IIOP. But the TM uses IIOP libs so they will be there regardless.
+* Remoting. It would be nice to be able to eliminate the need for remoting, e.g. for ejb3-lite if the user app isn't a remote ejb client. But that doesn't seem trivial and if a server is manageable remoting will be there anyway to support the CLI.
+* JCA stuff. But if there's a datasource it's there anyway.
+* Infinispan. A benefit of a future SLSB-only layer would be this dep could be eliminated. (But it might be used for web caching etc anyway.)
+* Remote artemis integration.  This we do save if ejb3-lite is used.
+
+
+== Test Plan
+
+The test coverage of the Galleon layers added by this proposal is divided in two main groups:
+
+. Test the Galleon layer provision.
+https://github.com/wildfly/wildfly/blob/master/testsuite/layers/src/test/java/org/jboss/as/test/layers/LayersTestCase.java[LayersTestCase]
+will be expanded to test the provision of the added layers when it is used isolated and with other layers.
+For each sort of provision, this test does the following:
+.. Verifies the provisioned modules are the expected ones.
+.. Verifies the provisioned server starts successfully.
+. Execution of existing tests on a minimum set of layers. Reuse the existing tests available on the WildFly test suite,
+which are directly testing features added by these layers, and execute them on a server installation provisioned with this layer.
+
+== Community Documentation
+
+Community documentation plan is adding the layer to https://docs.wildfly.org/19/Admin_Guide.html#wildfly-galleon-layers[WildFly Galleon layers]
```

Comment:
@chengfang the link should be to https://docs.wildfly.org/20/Admin_Guide.html#wildfly-galleon-layers